examples/larger-examples/datalog/ford-fulkerson.flix

```diff
@@ -0,0 +1,642 @@
+        let edges = inject g into Edge;
+        let rules = #{
+            Reach(x, y; init(y, x)) :- Edge(x, u, f, y),                 if (u - f) > 0. // Forward edge
+            Reach(x, z; cons(z, p)) :- Reach(x, y; p), Edge(y, u, f, z), if (u - f) > 0. // Forward edge
+            Reach(x, y; init(y, x)) :- Edge(y, u, f, x),                 if f > 0.       // Back edge
+            Reach(x, z; cons(z, p)) :- Reach(x, y; p), Edge(z, u, f, y), if f > 0.       // Back edge
+        };
+        let result = query edges, rules select fn from Reach(src, dst; fn);
+        Vector.head(result) |> Option.flatMap(toOption)
+
+    ///
+    /// Returns the most constraining capacity of `g` on the `Path` `p`.
+    ///
+    def minCapacity(p: Path[t], g: m[(t, Int32, Int32, t)]): Int32 with Foldable[m], Order[t] =
+        let onPath = (s, d) -> isForwardEdge(s, d, p) or isBackEdge(s, d, p);
+        let optMin = g |> Foldable.filter(match (s, _, _, d) -> onPath(s, d))
+            |> List.map(match (_, u, f, _) -> u - f)
+            |> List.minimum;
+        match optMin {
+            case Some(u) => u
+            case None    => unreachable!() // This function is only called by `maxFlow` if an augmenting path was found
+        }
+
+    ///
+    /// Returns a new flow network where the edges in `g` on the `Path` `p` has been adjusted by `incr`.
+    ///
+    def increaseFlow(p: Path[t], incr: Int32, g: m[(t, Int32, Int32, t)]): Vector[(t, Int32, Int32, t)] with Foldable[m], Order[t] =
+        g
+        |> Foldable.toVector
+        |> Vector.map(match (s, u, f, d) ->
+            if (isForwardEdge(s, d, p))
+                (s, u, f + incr, d)
+            else if (isBackEdge(s, d, p))
+                (s, u, f - incr, d)
+            else
+                (s, u, f, d)
+        )
+
+    ///
+    /// Returns true if `src` is an edge pointing to `dst` on the `Path` `p`.
+    ///
+    def isForwardEdge(src: t, dst: t, p: Path[t]): Bool with Eq[t] =
+        match (indexOf(src, p), indexOf(dst, p)) { // A path is sorted in reverse order
+            case (Some(si), Some(di)) if di + 1 == si => true
+            case _ => false
+        }
+
+    ///
+    /// Returns true if `dst` is an edge pointing to `src` on the `Path` `p`.
+    ///
+    def isBackEdge(src: t, dst: t, p: Path[t]): Bool with Eq[t] =
+        match (indexOf(src, p), indexOf(dst, p)) { // A path is sorted in reverse order
+            case (Some(si), Some(di)) if si + 1 == di => true
+            case _ => false
+        }
+
+    pub enum Path[a] with ToString {
+        case Path(List[a])
+        case Bot // Infinitely long path
+    }
+
+    instance Eq[Path[a]] {
+        pub def eq(x: Path[a], y: Path[a]): Bool = match (x, y) {
+            case (Bot, Bot)           => true
+            case (Bot, _)             => false
+            case (_, Bot)             => false
```

Comment:
Drop the two false cases and just make a default case at the bottom which is false.